.teamcity/Gradle_Check/configurations/FunctionalTest.kt

```diff
@@ -30,7 +30,7 @@ class FunctionalTest(model: CIBuildModel, testCoverage: TestCoverage, subProject
                             + buildScanValues.map { buildScanCustomValue(it.key, it.value) }
                     ).joinToString(separator = " "),
             timeout = testCoverage.testType.timeout,
-            daemon = useDaemon)
+            daemon = useDaemon && testCoverage.testType != TestType.noDaemon)
```

Comment:
I think we should do that decision in the model. I guess this time `FunctionTestProject` would be the way to go.
Something like
`val useDaemon = subProject.useDaemon && testConfig.testType != TestType.noDaemon`.

Reply:
The only problem is, `subProject` can be `null` here, because https://github.com/gradle/gradle/blob/69fc4f23dadde2716485239cb4905d03d02acf2f/.teamcity/Gradle_Check/projects/StageProject.kt#L60 this line won't pass `subProject` parameter at all, which makes the null-handling pretty annoying. Any ideas?

Reply:
`subProject` can't be null in `FunctionalTestProject`: https://github.com/gradle/gradle/blob/69fc4f23dadde2716485239cb4905d03d02acf2f/.teamcity/Gradle_Check/projects/FunctionalTestProject.kt#L16-L32
We only create the `buildType(...)` for `soak` tests - so `useDaemon` would be true there anyway, right?

Reply:
Thanks, I also did a refactoring work to avoid some repetition.